### PR TITLE
Exercises 1-4 Updates 

### PR DESCRIPTION
**Pull request by ksamreen**
*Issue #, if available:*

*Description of changes:*
-Updating code snippets in docs for 2-4 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/exercises/2-kms_encryption.rst [hunk 1]
```diff
@@ -439,35 +440,35 @@ View step-by-step changes in context, and compare your work if desired.
...
-                +        response = self.kms.decrypt(CiphertextBlob=ciphertext, EncryptionContext=encryption_context)
-                +        plaintext = response["Plaintext"]
-                +
-                         return json.loads(plaintext)
+                     :rtype: str
+                     """
+            +        encryption_context = {self._message_type: self._type_order_inquiry}
+                     plaintext = json.dumps(data).encode("utf-8")
+            -        return base64.b64encode(plaintext).decode("utf-8")
+            +        response = self.kms.encrypt(KeyId=self.key_id, Plaintext=plaintext, EncryptionContext=encryption_context)
```
[the remark this answers is not in the document]
Comment (ksamreen) [reply docs/exercises/2-kms_encryption.rst]: For 2, we set self.key_id = key_id_east, in the constructor so all the references following stay as key_id

File: docs/exercises/3-encryption_sdk.rst [hunk 2]
```diff
@@ -585,44 +584,43 @@ View step-by-step changes in context, and compare your work if desired.
...
-                +            if header.encryption_context[self._message_type] != self._type_order_inquiry:
-                +                raise KeyError()  # overloading KeyError to use the same exit whether wrong or missing
-                +        except KeyError:
-                +            raise ValueError("Bad message type in decrypted message")
-
-                         return json.loads(plaintext)
+                     """
+                     encryption_context = {self._message_type: self._type_order_inquiry}
+            -        plaintext = json.dumps(data).encode("utf-8")
+            -        response = self.kms.encrypt(KeyId=self.key_id, Plaintext=plaintext, EncryptionContext=encryption_context)
```
[the remark this answers is not in the document]
Comment (ksamreen) [reply docs/exercises/3-encryption_sdk.rst]: For 2, we leave set self.key_id = key_id_east, and then reference key_id everywhere. So, when this line is removed thats why it has the key_id reference.

File: docs/exercises/2-kms_encryption.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (lizroth) [reply docs/exercises/2-kms_encryption.rst]: Got it. Too much time looking at the Java one, I guess. Thanks!

File: docs/exercises/3-encryption_sdk.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (lizroth) [reply docs/exercises/3-encryption_sdk.rst]: Got it. Too much time looking at the Java one, I guess. Thanks!

File: docs/exercises/4-caching.rst [hunk 3]
```diff
@@ -120,13 +120,10 @@ Then, we'll replace our MasterKey field with a CryptoMaterialsManager:
...
 
-            master_key_provider = aws_encryption_sdk.KMSMasterKeyProvider(key_ids=[key_id])
+            master_key_provider = aws_encryption_sdk.KMSMasterKeyProvider(key_ids=[key_id_east])
             cache = aws_encryption_sdk.LocalCryptoMaterialsCache(capacity=100)
             self.materials_manager = aws_encryption_sdk.CachingCryptoMaterialsManager(
-                cache=cache,
-                master_key_provider=master_key_provider,
-                max_age=5.0 * 60.0,
-                max_messages_encrypted=10
+                cache=cache, master_key_provider=master_key_provider, max_age=5.0 * 60.0, max_messages_encrypted=10
```
[the remark this answers is not in the document]
Comment (ksamreen) [reply docs/exercises/4-caching.rst]: I am not sure why it is showing up like that here,  but when you build the html pages, it looks good.

File: docs/exercises/4-caching.rst [hunk 4]
```diff
@@ -362,31 +359,30 @@ View step-by-step changes in context, and compare your work if desired.
                      """Encrypt data.
             @@ -38,12 +43,12 @@ class EncryptDecrypt(object):
                      :returns: Base64-encoded, encrypted data
-                         :rtype: str
```
[the remark this answers is not in the document]
Comment (ksamreen) [reply docs/exercises/4-caching.rst]: This is weird, I don't know why its showing up like this in git, but at least when it renders it looks fine.

**Pull request merged**